ports/stm32/adc.c

```diff
@@ -659,8 +707,12 @@ float adc_read_core_vbat(ADC_HandleTypeDef *adcHandle) {
     // TEMPSENSOR or VREFINT works correctly.
     ADC->CCR &= ~ADC_CCR_VBATE;
     #endif
-
+    
+    #if defined(STM32H7)
+    return raw_value * VBAT_DIV * ADC_SCALE;
```

Comment:
Why isn't the adc_refcor needed for the H7 here?  Is it needed at all anywhere in adc.c?

Reply:
Again, sorry but not exactly sure why, it is needed for the temperature reading but not for vbat. If you add it you get a very high reading.

Reply:
As with core_vref, leave this alone until a proper fix is known.